pkg/object/objects_test.go

```diff
@@ -732,3 +732,417 @@ func TestUnmarshalWithJSONPB(t *testing.T) {
 		})
 	}
 }
+
+func TestManifestDiff(t *testing.T) {
+	testDeploymentYaml1 := `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: istio-citadel
+  namespace: istio-system
+  labels:
+    istio: citadel
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      istio: citadel
+  template:
+    metadata:
+      labels:
+        istio: citadel
+    spec:
+      containers:
+      - name: citadel
+        image: docker.io/istio/citadel:1.1.8
+        args:
+        - "--append-dns-names=true"
+        - "--grpc-port=8060"
+        - "--grpc-hostname=citadel"
+        - "--citadel-storage-namespace=istio-system"
+        - "--custom-dns-names=istio-pilot-service-account.istio-system:istio-pilot.istio-system"
+        - "--monitoring-port=15014"
+        - "--self-signed-ca=true"
```

Comment:
having this much stuff in the resource actually makes it harder to know what the test is doing and what is expected. i recommend defining very minimal resources so it's clear what the differences are.